runtime/signals.c

```diff
@@ -350,56 +349,56 @@ value caml_do_pending_actions_exn(void)
   Caml_state->action_pending = 0;
 
   /* Call signal handlers first */
-  value exn = caml_process_pending_signals_exn();
-  if (Is_exception_result(exn)) goto exception;
+  caml_result result = caml_process_pending_signals_result();
+  if (result.is_exception) goto exception;
 
 #if 0
   /* Call memprof callbacks */
-  exn = caml_memprof_handle_postponed_exn();
-  if (Is_exception_result(exn)) goto exception;
+  result = caml_memprof_handle_postponed_result();
+  if (result.is_exception) goto exception;
 #endif
 
   /* Call finalisers */
-  exn = caml_final_do_calls_exn();
-  if (Is_exception_result(exn)) goto exception;
+  result = caml_final_do_calls_result();
+  if (result.is_exception) goto exception;
 
   /* Process external interrupts (e.g. preemptive systhread switching).
      By doing this last, we do not need to set the action pending flag
      in case a context switch happens: all actions have been processed
      at this point. */
   caml_process_external_interrupt();
 
-  return Val_unit;
+  return result;
```

Comment:
```suggestion
  return Result_unit;
```
Clearer and more robust: be faithful to the original rather than rely on the fact that `caml_final_do_calls_result` happens to return `Result_unit`.